fuel-asm/src/macros.rs

```diff
@@ -286,6 +286,73 @@ macro_rules! impl_instructions {
         }
     };
 
+    // Recursively generate a test constructor for each opcode
+    (impl_opcode_test_construct $doc:literal $ix:literal $Op:ident $op:ident [$($field:ident)*] $($rest:tt)*) => {
+        impl crate::_op::$Op {
+            impl_instructions!(impl_opcode_test_construct_fn [$($field)*]);
+        }
+        impl_instructions!(impl_opcode_test_construct $($rest)*);
+    };
+    (impl_opcode_test_construct) => {};
+
+    (impl_opcode_test_construct_fn [RegId]) => {
+        #[doc(hidden)]
+        /// Construct the instruction from all possible raw fields, ignoring inapplicable ones.
+        pub fn test_construct(ra: RegId, _rb: RegId, _rc: RegId, _rd: RegId, _imm: u32) -> Self {
+            Self(pack::bytes_from_ra(ra))
+        }
```

Comment:
If we're only using these for testing, maybe we should conservatively add them behind `#[cfg(test)]`?

Reply:
We would have to change the encoding test to be an unit test, as integration tests cannot use `#[cfg(test)]` features

Reply:
Ahh in that case, maybe we would be better off moving these tests into unit tests? I'm a bit weary of exposing this constructor in our public API is all - ideally we'd only expose stuff we expect downstream to actually use.

Reply:
doesn't really seem like the encoding tests need to be integ tests anyways